### PR TITLE
Add  API Reference documentation as Github pages

### PR DESCRIPTION
**Pull request by robin-ynput**
## Changelog Description
Addresses #21 

* Implemented sphinx autodoc as ci workflow
* Few docstring edits to remove sphinx errors
* Sphinx configuration polishes

**Result here : https://ynput.github.io/ayon-python-api/build/html/** (temporary url)

## Additional info
Potential follow-up issues not covered by this PR:
* #171 
* https://github.com/ynput/ayon-documentation/issues/215

### CODE REVIEW

File: ayon_api/_api.py [hunk 1]
```diff
@@ -4049,12 +4053,12 @@ def get_entities_links(*args, **kwargs):
     Args:
         project_name (str): Project where links are.
         entity_type (Literal["folder", "task", "product",
-            "version", "representations"]): Entity type.
+        |    "version", "representations"]): Entity type.
```
Comment (robin-ynput) [inline ayon_api/_api.py:4056]: `|` are for sphinx to not complain about the additional indent. Not too sure why it complained here, seems like a glitch to me.

**Comment by robin-ynput** [comment]
@jakubjezek001 I could reproduce but couldn't find any fix regarding the style hiccup you mentioned. 
gif_anim_capture

This definitely comes from the theme CSS .
* -webkit-font-smoothing: antialiased!important;
* 1rem


It could be related to the page content being quite long to format... anyway, this is very inconsistent.
If that's an issue, I can probably inject a custom CSS sheet to prevent this.

File: ayon_api/_api.py (hunk 1 above)
Comment (iLLiCiTiT) [reply ayon_api/_api.py:4056]: It has to be changed in `server_api.py`, content of `_api.py` is created based on that.

File: ayon_api/server_api.py [hunk 2]
```diff
@@ -7785,13 +7789,10 @@ def get_entities_links(
 
         Args:
             project_name (str): Project where links are.
-            entity_type (Literal["folder", "task", "product",
-                "version", "representations"]): Entity type.
-            entity_ids (Optional[Iterable[str]]): Ids of entities for which
-                links should be received.
+            entity_type (Literal["folder", "task", "product", "version", "representations"]): Entity type.
+            entity_ids (Optional[Iterable[str]]): Ids of entities for which links should be received.
```
Comment (iLLiCiTiT) [inline ayon_api/server_api.py:7793]: Too long lines.

Comment (robin-ynput) [reply ayon_api/server_api.py]: Adjusted (in the code and in my IDE), thanks :)

File: ayon_api/_api.py [hunk 3]
```diff
@@ -4028,7 +4029,10 @@ def delete_link(*args, **kwargs):
 def get_entities_links(*args, **kwargs):
     """Helper method to get links from server for entity types.
 
-    Example output::
+    .. highlight:: text
+    .. code-block:: text
+
+        Example output
```
Comment (iLLiCiTiT) [inline ayon_api/_api.py:4035]: Should this be without colon `:` (differs from docstring in `ServerAPI`)?

Comment (robin-ynput) [reply ayon_api/_api.py]: Thanks for the tip, and sorry for the back and forth.

**Review by iLLiCiTiT — APPROVED** [review]

**Pull request merged**